operator/pkg/hook/hook.go

```diff
@@ -0,0 +1,117 @@
+package hook
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/megaease/easemesh/mesh-operator/pkg/base"
+	"github.com/megaease/easemesh/mesh-operator/pkg/deploymentmodifier"
+	"github.com/pkg/errors"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	v1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	annotationPrefix              = "mesh.megaease.com/"
+	annotationServiceNameKey      = annotationPrefix + "service-name"
+	annotationAppContainerNameKey = annotationPrefix + "app-container-name"
+	annotationApplicationPortKey  = annotationPrefix + "application-port"
+	annotationAliveProbeURLKey    = annotationPrefix + "alive-probe-url"
+)
+
+type (
+	// MutateHook handle requests from the injector MutatingWebhookConfiguration.
+	MutateHook struct {
+		*base.Runtime
+		Admission *webhook.Admission
+	}
+)
+
+// NewMutateHook creates a mutate hook.
+func NewMutateHook(baseRuntime *base.Runtime) *MutateHook {
+	h := &MutateHook{
+		Runtime: baseRuntime,
+	}
+	h.Admission = &webhook.Admission{
+		Handler: admission.HandlerFunc(h.mutateHandler),
+	}
+	h.Admission.InjectLogger(h.Log)
+
+	return h
+}
+
+func (h *MutateHook) mutateHandler(cxt context.Context, req admission.Request) admission.Response {
+	switch req.Operation {
+	case admissionv1.Connect, admissionv1.Delete:
+		return ignoreResp(req)
+	}
+
+	if req.Kind.Kind != "Deployment" {
```

Comment:
Overall, the implementation is good to me, it should be better if we mutated pod instead of Deployment, which means that we could support Deployment, Statefulset, DaemonSet at the same time in one implementation.